clang/lib/AST/Interp/EvaluationResult.cpp

```diff
@@ -122,19 +123,22 @@ static bool CheckFieldsInitialized(InterpState &S, SourceLocation Loc,
   }
 
   // Check Fields in all bases
-  for (const Record::Base &B : R->bases()) {
+  for (unsigned I = 0, E = R->getNumBases(); I != E; ++I) {
+    const auto &B = *R->getBase(I);
     Pointer P = BasePtr.atField(B.Offset);
     if (!P.isInitialized()) {
       const Descriptor *Desc = BasePtr.getDeclDesc();
-      if (Desc->asDecl())
-        S.FFDiag(BasePtr.getDeclDesc()->asDecl()->getLocation(),
-                 diag::note_constexpr_uninitialized_base)
+      if (const auto *CD = dyn_cast_or_null<CXXRecordDecl>(R->getDecl())) {
+        const auto &BS = *std::next(CD->bases_begin(), I);
```

Comment:
Does `CD->bases()[I]` not work?

Reply:
Yeah, the compiler says "Type 'clang::CXXRecordDecl::base_class_const_range' (aka 'iterator_range<const clang::CXXBaseSpecifier *>') does not provide a subscript operator".